src/features/Bluetooth/Program/ProgramSlice.ts

```diff
@@ -123,8 +125,10 @@ export function* tryProgramNewSensor({
       yield retry(10, 0, btService.toggleButton, macAddress);
     }
 
+    const startOfMinute = utils.startOfMinute(logDelay);
```

Comment:
main change is here, which is calculating the start of the minute here rather than in a component